qlever.py

```diff
@@ -0,0 +1,488 @@
+                "no_patterns": "no",
+            },
+            "docker": {
+                "image": "adfreiburg/qlever",
+                "container_server": f"qlever.server.{self.name}",
+                "container_indexer": f"qlever.indexer.{self.name}",
+            },
+            "ui": {
+                "port": "7000",
+            }
+        }
+        for section in defaults:
+            for option in defaults[section]:
+                if not self.config[section].get(option):
+                    self.config[section][option] = defaults[section][option]
+
+        # Show some information (for testing purposes only).
+        print(f"Parsed Qleverfile, sections are: "
+              f"{', '.join(self.config.sections())}")
+        print(f"Name of dataset: {self.config['DEFAULT']['name']}")
+        # print(f"Get the data: {self.config['data']['get_data_cmd']}")
+
+        # Check specifics of the installation.
+        self.check_installation()
+
+    def check_installation(self):
+        """ Several checks which are important for the script to work properly.
+        """
+
+        # Handle the case Systems like macOS do not allow
+        # psutil.net_connections().
+        try:
+            psutil.net_connections()
+            self.net_connections_enabled = True
+        except Exception as e:
+            self.net_connections_enabled = False
+            if self.log_level_is_debug():
+                print(f"Note: psutil.net_connections() failed ({e}),"
+                      f" will not scan network connections for action"
+                      f" \"start\"")
+
+        # Check whether docker is installed and works (on MacOS 12, docker
+        # hangs when installed without GUI, hence the timeout).
+        try:
+            completed_process = subprocess.run(
+                    ["docker", "info"], timeout=0.5,
+                    stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+            if completed_process.returncode != 0:
+                raise Exception("docker info failed")
+            self.docker_enabled = True
+        except Exception:
+            self.docker_enabled = False
+            print("Note: `docker info` failed, therefore"
+                  " docker.USE_DOCKER=true not supported")
+
+        # Check if the QLever binaries work.
+        try:
+            subprocess.run([self.config['server']['binary'], "--help"],
+                           stdout=subprocess.DEVNULL,
+                           stderr=subprocess.DEVNULL)
+            subprocess.run([self.config['index']['binary'], "--help"],
+                           stdout=subprocess.DEVNULL,
+                           stderr=subprocess.DEVNULL)
+            self.binaries_work = True
+        except Exception:
+            print("Note: QLever binaries not found or failed, therefore"
+                  " docker.USE_DOCKER=false not supported")
+            self.binaries_work = False
+
```

Comment:
Should we immediately terminate the script if we have neither a working Docker nor a working native binary setup? I think this would be the right position to give a useful error message here.

Also: the `self.binaries_work` variable is set here, but never used as far as I see.